core/src/test/scala/kafka/tools/GetOffsetShellTest.scala

```diff
@@ -109,6 +111,46 @@ class GetOffsetShellTest extends KafkaServerTestHarness with Logging {
     )
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = Array("-1", "latest"))
+  def testGetLatestOffsets(time: String): Unit = {
+    val offsets = executeAndParse(Array("--topic-partitions", "topic.*:0", "--time", time))
+    assertEquals(
+      List(
+        ("topic1", 0, Some(1)),
+        ("topic2", 0, Some(2)),
+        ("topic3", 0, Some(3)),
+        ("topic4", 0, Some(4))
+      ),
+      offsets
+    )
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = Array("-2", "earliest"))
+  def testGetEarliestOffsets(time: String): Unit = {
+    val offsets = executeAndParse(Array("--topic-partitions", "topic.*:0", "--time", time))
+    assertEquals(
+      List(
+        ("topic1", 0, Some(0)),
+        ("topic2", 0, Some(0)),
+        ("topic3", 0, Some(0)),
+        ("topic4", 0, Some(0))
+      ),
+      offsets
+    )
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = Array("-3", "max-timestamp"))
+  def testGetOffsetsByMaxTimestamp(time: String): Unit = {
+    val offsets = executeAndParse(Array("--topic-partitions", "topic.*", "--time", time))
+    offsets.foreach{ offset =>
```

Comment:
nit: Could we deconstruct offset? `offsets.foreach { case (topic, partition, timestamp) ->`

Reply:
Good suggestion